### PR TITLE
[BACKLOG-35000] Adding com.hitachivantara.security.web packages to custom.properties

### PR DESCRIPTION
**Pull request by dcleao**
Please, review:
@graimundo
/cc @bennychow if you want to review, be my guest!
/cc @jcarneirohv

Depends on pentaho/security-web#4.

### CODE REVIEW

File: assemblies/common-resources/src/main/resources-filtered/etc/custom.properties [hunk 1]
```diff
@@ -184,7 +184,8 @@ org.osgi.framework.system.packages.extra= \
  org.h2.*, \
  com.cloudera.impala.jdbc41, \
  org.postgresql.*, \
- org.pentaho.di.core.namedcluster.*;version="${big-data-plugin.version}"
+ org.pentaho.di.core.namedcluster.*;version="${big-data-plugin.version}", \
+ com.hitachivantara.security.web.*;version="${hv-security-web.version}"
```
Comment (graimundo) [inline assemblies/common-resources/src/main/resources-filtered/etc/custom.properties:188]: Changes are only for pentaho server, remove the package export from here as this affects other assemblies such as PDI.

File: assemblies/server/src/main/resources-filtered/etc/custom.properties [hunk 2]
```diff
@@ -159,7 +159,8 @@ org.osgi.framework.system.packages.extra= \
  org.apache.hbase.thirdparty.*, \
  org.apache.hbase.thirdparty.com.google.common.primitives.*, \
  com.google.common.collect;version="12.0.0", \
- org.pentaho.di.core.namedcluster.*;version="${big-data-plugin.version}"
+ org.pentaho.di.core.namedcluster.*;version="${big-data-plugin.version}", \
+ com.hitachivantara.security.web.*;version="${hv-security-web.version}"
```
Comment (graimundo) [inline assemblies/server/src/main/resources-filtered/etc/custom.properties:163]: As per our conversation in slack, check if its possible to only export the API packages.

**Review by graimundo — CHANGES_REQUESTED** [review]

**Comment by dcleao** [comment]
Rebased into master.

**Comment by dcleao** [comment]
Addressed code review comments.

**Comment by dcleao** [comment]
Had to revert the export "api.*" only change. The reason being the `pentaho / data-refinery-pdi-plugin` OSGi bundle, which is installed in the Pentaho server. This bundle uses the CSRF JAXRS-v1 client, in the `ModelServerPublish` action, which is only really used when this bundle is installed in PDI. Nonetheless, OSGi bootstrap would fail if the `com.hitachivantara.security.web / csrf-token-service-client-java-jax-rs-v1` **bundle** wouldn't be available.

Making both the `api` and `impl` packages be exported from the root bundle is an easy, quick fix to this problem.
We can discuss later whether we should:
1. Keep this solution
2. Change the data refinery plugin to include the impl dependency (Maven `<scope>compile</scopoe>`).
3. Create a Karaf feature with the impl bundles of the Security Web library.

**Comment by dcleao** [comment]
Reverted the previous revert, choosing to add the implementation dependency to the PDI data refinery feature instead.

**Review by graimundo — APPROVED** [review]

**Pull request merged**